breakpoints/getCallbackFromUserFriendlyCallbackArgument.js

```diff
@@ -50,7 +50,31 @@ function getTraceFunction(predefinedBreakpoint) {
     else {
         traceFn = function(debugInfo){
             runWithBreakpointsDisabled(function(){
-                console.trace("About to " + debugInfo.accessType + " property '" + debugInfo.propertyName + "' on this object: ", debugInfo.object)
+                if (debugInfo.accessType == "set") {
+                    var newPropertyValue = debugInfo.newPropertyValue;
+                    var newPropertyType = typeof newPropertyValue;
+
+                    if (newPropertyType === "string" || newPropertyType === "number") {
+                        newPropertyValue = newPropertyValue.toString();
+
+                        if (newPropertyValue.length > 10) {
+                            newPropertyValue = newPropertyValue.substring(0, 10) + "...";
+                        }
+
+                        if (newPropertyType === "string") {
+                            newPropertyValue = "'" + newPropertyValue + "'";
+                        }
+                    }
+                    else if (newPropertyType !== "boolean") {
```

Comment:
It took me a while to understand what's going on here. It wasn't clear to me where the `boolean` case is being handled.

Can you break it up so it's

```
if (string or number) {

}
else if (boolean) {

} else {

}
```